spec/syntax.md

```diff
@@ -740,6 +747,20 @@ when matching _name_ or _identifier_ strings or _unquoted literal_ values.
 
 _Variable_ _names_ are prefixed with `$`.
 
+Two _names_ are considered equal if they are canonically equivalent strings,
+that is, if they consist of the same sequence of Unicode code points after
+[Unicode Normalization Form C](https://unicode.org/reports/tr15/) ("NFC")
+has been applied to both.
+
+> [!NOTE]
+> Implementations are not required to normalize _names_.
```

Comment:
Should clarify, if/as we're normalising key values.
```suggestion
> Implementations are not required to normalize all _names_.
```

Reply:
Keys are not names?

```abnf
key               = literal / "*"
```

Reply:
They can be when you go down deeper.
```abnf
literal = quoted-literal / unquoted-literal
unquoted-literal = name / number-literal
```

Reply:
It doesn't really mean "name" there, though. I think that's us saving effort by defining what can be unquoted by recycling `name`. Ignoring the quoted/unquoted part syntactical representation, a key is just a literal value.

Reply:
Somewhat, yes, but we do say this: https://github.com/unicode-org/message-format-wg/blob/95ec6d53220d7b61c70ebef5647d65592f3e2e62/spec/syntax.md?plain=1#L732-L734

Rather than redefining _name_, I think it'd be easier to just include the "all" here. That way we don't need to explain how sometimes a `name` is not a _name_.

Reply:
My reluctance here is that "all" suggests that implementations might be required to normalize some names that aren't keys. In fact, not even keys are required to be stored in NFC. But it's such a small point that I'll commit your suggestion now.